AudioLink/Shaders/AudioLink.shader

```diff
@@ -319,8 +319,31 @@ Shader "AudioLink/Internal/AudioLink"
 
                 // If part of the delay
                 } else {
-                    // Return pixel to the left
-                    return AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(coordinateLocal.x - 1, coordinateLocal.y));
+                    // Slide pixels (coordinateLocal.x > 0)
+                    float4 lastval_timing = AudioLinkGetSelfPixelData(ALPASS_GENERALVU + int2(4, 1)); // Timing for 4-band, move at 90 Hz.
+                    lastval_timing.x += unity_DeltaTime.x * AUDIOLINK_4BAND_TARGET_RATE;
+                    int frames_to_roll = floor( lastval_timing.x );
+
+                    if( frames_to_roll == 0 )
+                    {
+                        return AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(coordinateLocal.x, coordinateLocal.y));
+                    }
+                    else // 1 or more.
+                    {
+                        if( coordinateLocal.x > frames_to_roll )
+                        {
+                            // For the rest of the line, move by the appropriate speed
+                            return AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(coordinateLocal.x - frames_to_roll, coordinateLocal.y));
+                        }
+                        else
+                        {
+                            // For the first part, extrapolate the cells.
+                            float last = AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(0, coordinateLocal.y));
+                            float next = AudioLinkGetSelfPixelData(ALPASS_AUDIOLINK + int2(1, coordinateLocal.y));
+                            float lperv = (coordinateLocal.x - 1) / (float)frames_to_roll;
```

Comment:
is 'lperv' a typo here? what is the abbreviation

Reply:
Prev is correct.  will address.